website_hr_contact/controllers/main.py

```diff
@@ -0,0 +1,155 @@
+
+import werkzeug
+
+from openerp import http
+from openerp.addons.website.models.website import slug
+from openerp.osv.orm import browse_record
+
+
+class QueryURL(object):
+    def __init__(self, path='', path_args=None, **args):
+        self.path = path
+        self.args = args
+        self.path_args = set(path_args or [])
+
+    def __call__(self, path=None, path_args=None, **kw):
+        path = path or self.path
+        for k, v in self.args.items():
+            kw.setdefault(k, v)
+        path_args = set(path_args or []).union(self.path_args)
+        paths, fragments = [], []
+        for key, value in kw.items():
+            if value and key in path_args:
+                if isinstance(value, browse_record):
+                    paths.append((key, slug(value)))
+                else:
+                    paths.append((key, value))
+            elif value:
+                if isinstance(value, list) or isinstance(value, set):
+                    fragments.append(
+                        werkzeug.url_encode([(key, item) for item in value]))
+                else:
+                    fragments.append(werkzeug.url_encode([(key, value)]))
+        for key, value in paths:
+            path += '/' + key + '/%s' % value
+        if fragments:
+            path += '?' + '&'.join(fragments)
+        return path
+
+
+class WebsiteHrAddressbook(http.Controller):
+    _items_per_page = 20
+
+    def _get_children_department_recursive(self, root, visible_departments):
+        tree = []
+        for department in root.child_ids:
+            children = self._get_children_department_recursive(
+                department, visible_departments)
+            if department in visible_departments:
+                tree.append({
+                    'department': visible_departments.filtered(
+                        lambda rec: rec.id == department.id),
+                    'children': children
+                    })
+            else:
+                tree.extend(children)
+        return tree
+
+    def _get_departments_tree(self, department=None, search=None, **post):
+        """ Build a department tree with only published departments
+        The tree only contains published department even if the parent
+        department is not published.
+        """
+        tree = []
+        request = http.request
+        hr_department = request.env['hr.department']
+        sudo_hr_department = hr_department.sudo()
+        visible_departments = hr_department.search([(1, '=', 1)])
+        departments = sudo_hr_department.search([('parent_id', '=', False)])
+        for department in departments:
+            children = self._get_children_department_recursive(
+                department, visible_departments)
+            if department in visible_departments:
+                tree.append({
+                    'department': visible_departments.filtered(
+                        lambda rec: rec.id == department.id),
```

Comment:
@andreparames `department` is a department in the `sudo`Environment. By filtering visible_departments I take the department in the user's Environement.